### PR TITLE
Stop FileSystem WriteAsync test from writing too much data to disk

### PR DESCRIPTION
**Pull request by wtgodbe**
For https://github.com/dotnet/corefx/issues/24505

@danmosemsft PTAL

### CODE REVIEW

File: src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs [hunk 1]
```diff
@@ -418,7 +418,7 @@ public Task ManyConcurrentWriteAsyncs()
                         }
                     } while (!tokenSource.Token.IsCancellationRequested);
                 }
-            } while (DateTime.UtcNow - testStartTime <= testRunTime);
+            } while (DateTime.UtcNow - testStartTime <= testRunTime && totalBytesWritten < 10000000);
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs]: Doesn't this sort of defeat the purpose of the test?  How long does the test take now with this limitation?

Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs]: where is `totalBytesWritten ` assigned?

Comment (stephentoub) [reply src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs]: Actually, I'm not sure this even addresses the concern. All the writing is happening in the inner loop. By the time we get here, we've written everything to the file we're going to write, and when we loop around the outer loop, it overwrites the file with a new empty one to start again.

I can see the value of this condition, but i think for it to be meaningful it needs to be in the inner loop. And then some we're using a length limitation, we can remove the time limitation enabled by the CancellationToken.

Comment (danmosemsft) [reply src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs]: Nit, 10_000_000 is easier to read than 10000000

**Comment by ahsonkhan** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by wtgodbe** [comment]
@dotnet-bot help

**Comment by wtgodbe** [comment]
@dotnet-bot test Outerloop Linux x64 Release Build
@dotnet-bot test Outerloop OSX x64 Release Build
@dotnet-bot test Outerloop NETFX x64 Release Build
@dotnet-bot test Outerloop Windows x64 Release Build

File: src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs]: Well the CancelationToken would still provide value -- it means we need not write all the bytes, and it randomizes somewhat. But fine.

**Comment by wtgodbe** [comment]
@danmosemsft I don't have a problem with adding the cancellation token back, I was under the impression it was only needed for timing, which was why I removed it

**Comment by danmosemsft** [comment]
Up to you. This isn't worth much more time investment

**Comment by wtgodbe** [comment]
I had to fix a syntax error anyways, so I re-added it in the most recent commit.

**Comment by wtgodbe** [comment]
@dotnet-bot test Outerloop Linux x64 Release Build
@dotnet-bot test Outerloop OSX x64 Release Build
@dotnet-bot test Outerloop NETFX x64 Release Build
@dotnet-bot test Outerloop Windows x64 Release Build

**Comment by danmosemsft** [comment]
Process failures are https://github.com/dotnet/corefx/pull/27557
HTTP is https://github.com/dotnet/corefx/issues/26092

**Pull request merged**